podcast_agent/client/transcriber.py

```diff
@@ -32,14 +33,49 @@ def read_from_url(self, url):
                     temp_file.write(audio_file.content)
                     temp_file_name = temp_file.name
                     temp_file.seek(0)
-                    transcript = self.local_file_to_transcription(open(temp_file_name, 'rb'))
+
+                    song = AudioSegment.from_mp3(temp_file.name)
+                    ten_minutes = 10 * 60 * 1000
+                    segments = self.get_segments(song, 10**5)
+                    [segment.export(temp_file.name + f'_{i}' for i, segment in enumerate(segments))]
+
+                    transcript_segments = [self.local_file_to_transcription(open(temp_file.name + f'_{i}', 'rb')) for  i in range(len(segments)) ]
+
+                    transcript = ''.join(transcript_segments)#self.local_file_to_transcription(open(temp_file_name, 'rb'))
 
         return transcript
 
-    def local_file_to_transcription(self, fileToTranscribe):
+    def get_segments(self, L, mSS):
+        return [L[i:(i + 1)*mSS] for i in range(len(L)//mSS) + 1]
+
+
+    def local_file_to_transcription(self, path_to_file_to_transcribe):
         #call openai transcription API on given file
         #
-        #fileToTranscribe: file like object
+        #path_to_file_to_transcribe: path to a file
         #returns: text transcription of file
+        import os
+        
+        if not os.path.isfile(path_to_file_to_transcribe):
+            raise FileNotFoundError(f'[Errno 2] no such file {path_to_file_to_transcribe}')
+
+        audio = AudioSegment.from_mp3(path_to_file_to_transcribe)
+        prefix = path_to_file_to_transcribe.replace('.mp3', '')
+        mSS = 6*(10**5)
+        print(1 + (len(audio)//mSS))
+
+        def write_temp_file(abs_path, content):
+            with open(abs_path, 'wb') as target:
+                target.write(content)
+        
+        num_threads = os.cpu_count()
+
```

Comment:
based on whether calling `export` on an audio segment is compute, ie generating the valid mp3, or IO intensive, parallelizing over processes or threads may help with speed (will look into which strategy is best in this context), but we also need to avoid having too many files open at once

Reply:
next steps PRs: 

1. add support to read/write transcriptions from supabase
2. optimize mp3 segmentation export